src/razor/src/csharp/csharpProjectedDocument.ts

```diff
@@ -80,6 +82,7 @@ export class CSharpProjectedDocument implements IProjectedDocument {
         if (this.provisionalEditAt && this.preProvisionalContent) {
             // Undo provisional edit if one was applied.
             this.setContent(this.preProvisionalContent);
+            this.resolveProvisionalEditAt = this.provisionalEditAt;
```

Comment:
Perhaps this field should be cleared in `addProvisionalDotAt`, just to ensure the previous resolve index doesn't stick around?

Reply:
The problem here is right now the resolve completion request doesn't know whether the completion request was provisional or not. So one bad state could be: 
-> (Provisional completion) user triggers provisional completion
-> (resolve completion) provisional dot are added and removed as the user toggles completion list 
-> (regular completion) as the user continues to type, completion list shows up 
-> (resolve completion) provisional dot are added and removed still since the field is not cleared